public/script.js

```diff
@@ -5431,70 +5431,91 @@ export function setSendButtonState(value) {
     is_send_press = value;
 }
 
-async function renameCharacter() {
-    const oldAvatar = characters[this_chid].avatar;
-    const newValue = await callPopup('<h3>New name:</h3>', 'input', characters[this_chid].name);
+export async function renameCharacter(name = null, { silent = false, renameChats = null } = {}) {
+    if (!name && silent) {
+        toastr.warning('No character name provided.', 'Rename Character');
+        return false;
+    }
 
-    if (newValue && newValue !== characters[this_chid].name) {
-        const body = JSON.stringify({ avatar_url: oldAvatar, new_name: newValue });
-        const response = await fetch('/api/characters/rename', {
-            method: 'POST',
-            headers: getRequestHeaders(),
-            body,
-        });
+    const oldAvatar = characters[this_chid].avatar;
```

Comment:
Whoops, missed that the rename method didn't have that check before. Likely because it was only used by the button that was hidden otherwise.

Done.